### PR TITLE
Added impl to update document parties and shipment locations.

### PR DESCRIPTION
**Pull request by douradofrazer**
Resolves:
https://dcsa.atlassian.net/browse/DDT-506

### CODE REVIEW

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java [hunk 1]
```diff
@@ -763,25 +753,31 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
...
-      LocationTO locationTO, Function<String, Mono<Boolean>> updateBookingCallback) {
+      String currentLocationIDInBooking,
+      LocationTO locationTO,
+      Function<String, Mono<Boolean>> updateBookingCallback) {
 
+    // locationTO is the location received from the update booking request
     if (Objects.isNull(locationTO)) {
-      return Mono.just(Optional.empty());
+      if (StringUtils.isEmpty(currentLocationIDInBooking)) {
+        // it's possible that there may be no location liked to booking
```
Comment (mp-assco) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:763]:
```suggestion
        // it's possible that there may be no location linked to booking
```

typo.

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java [hunk 2]
```diff
@@ -146,6 +146,8 @@ private ShipmentSummaryTO createShipmentSummaryTO(Shipment shipment, Booking boo
   public Mono<BookingTO> createBooking(final BookingTO bookingRequest) {
 
     Booking requestedBooking = bookingMapper.dtoToBooking(bookingRequest);
+    // CarrierBookingRequestReference is not allowed to be set by request
+    requestedBooking.setCarrierBookingRequestReference(null);
```
Comment (pimg) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:150]: could this be done by adding @Setter(AccessLevel.NONE) to the BookingTO?

**Review by pimg — CHANGES_REQUESTED** [review]
could the explicit set to null be implemented directly on the BookingTO?

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java (hunk 2 above)
Comment (douradofrazer) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:150]: but then that would prevent a request from PUT from settings it as well,  we accept the CarrierBookingRequestReference  both in path and body and we need to validate that they are the same for a PUT request.

**Review by pimg — APPROVED** [review]

**Review by pimg — APPROVED** [review]

**Pull request merged**